services/repository/src/main/java/org/sagebionetworks/repo/web/controller/EntityController.java

```diff
@@ -266,13 +266,15 @@
 	 */
 	@ResponseStatus(HttpStatus.CREATED)
 	@RequestMapping(value = { UrlHelpers.ENTITY }, method = RequestMethod.POST)
-	public @ResponseBody Entity createEntity(@RequestParam(value = AuthorizationConstants.USER_ID_PARAM) Long userId,
+	public @ResponseBody Entity createEntity(
+			@RequestHeader(value = AuthorizationConstants.SYNAPSE_AUTHORIZATION_HEADER_NAME, required=true) String authorizationHeader,
```

Comment:
We can consider using a HandlerMethodArgumentResolver (https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/web/method/support/HandlerMethodArgumentResolver.html) so that we can inject and resolve a parameter by its type in the case when we have a parsed token or alternatively a UserInfo object. Also see https://www.petrikainulainen.net/programming/spring-framework/spring-from-the-trenches-creating-a-custom-handlermethodargumentresolver.

Reply:
Done.  See the newly added file: services/repository/src/main/java/org/sagebionetworks/repo/web/UserInfoMethodArgumentResolver.java